staging/src/k8s.io/apiserver/pkg/storage/cacher/lister_watcher.go

```diff
@@ -59,7 +63,11 @@ func (lw *listerWatcher) List(options metav1.ListOptions) (runtime.Object, error
 		Predicate:            pred,
 		Recursive:            true,
 	}
-	if err := lw.storage.GetList(context.TODO(), lw.resourcePrefix, storageOpts, list); err != nil {
+	ctx := context.Background()
+	if lw.contextMetadata != nil {
```

Comment:
This is more of a question than a suggestion, but considering that we are now changing the signature of a public function (`NewListWatcher`), for folks using this in custom API servers etc, do you think we should allow passing custom `contextMetadata` through this? I'm not sure if this would lead to behaviour that we define or even recommend.

Maybe what I had in mind was something like
```go
if lw.contextMetadata != nil {
    mdValue := lw.contextMetadata.Get("source")
    if len(md) > 0 && md[0] == "cache" {
      ctx = metadata.NewOutgoingContext(ctx, lw.contextMetadata)
    }
}
```